### PR TITLE
Add the option `--features`

### LINKED ISSUE #1647: Introduce a facility for enabling experimental features
As described in #401, we need a flag to enable experimental features. For instance, `--feature=rows`.

### PR DESCRIPTION
**Pull request by konnov**
Closes #1647. As discussed in #401, this PR introduces the option `--features` that lets us pass experimental features to the tool. The only supported feature in `rows`, which is parsed as `RowsFeature`. When we want to add new features, we have to extend the trait `Feature` with a case class and update the parser in `General`.

For the moment, I am not adding any documentation to the user manual. I will extend the manual section on types, when we start the transition period to Type System 1.2.

- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] Documentation added for any new functionality
- [x] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

### CODE REVIEW

**Review by thpani — COMMENTED** [review]
Code LGTM, but I think we should discuss how this will evolve once the feature becomes stable. Do you have a plan for that?

File: mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/General.scala
Comment (thpani) [reply mod-tool/src/main/scala/at/forsyte/apalache/tla/tooling/opt/General.scala:39]: Resolving because we discussed this offline

**Review by thpani — APPROVED** [review]
LGTM (modulo fixing the output on that failing integration test)

**Review by shonfeder — CHANGES_REQUESTED** [review]
Leaving a note that I have requested a change set via #1652

**Comment by thpani** [comment]
> Leaving a note that I have requested a change set via #1652

@shonfeder quick workflow question: I think we don't necessarily need to block on this?
It should be fine to review #1652 against this branch, then rebase it on `unstable` once this PR gets merged?

**Review by shonfeder — APPROVED** [review]

**Pull request merged**
